native/Avalonia.Native/src/OSX/common.h

```diff
@@ -89,6 +89,13 @@ template<typename T> class ObjCWrapper {
 - (void) action;
 @end
 
+@implementation NSScreen (AvNSScreen)
+- (CGDirectDisplayID)av_displayId
+{
+    return [self.deviceDescription[@"NSScreenNumber"] unsignedIntValue];
```

Comment:
> Also, why do we need our own NSScreen implementation class?

Well, I copy pasted this technique from qtbase code.
https://github.com/qt/qtbase/blob/213127a013b6ecafb36eae1014d5988198736712/src/plugins/platforms/cocoa/qcocoascreen.mm#L797-L804

Although I just found another one a little bit different
https://github.com/qt/qtbase/blob/213127a013b6ecafb36eae1014d5988198736712/src/plugins/platforms/cocoa/qcocoascreen.h#L114-L119

Reply:
> Not sure if we need/want to do something about that.

With current implementation, it will destroy old screen, and create another one. It will be another instance but should be fine in general.
I wonder though, if NSScreen instance by itself is preserved between GPU switches.

Reply:
> Also, why do we need our own NSScreen implementation class? 

Seems to work fine. Can double check though.